pkg/controllers/utils/poddecoration/sort.go

```diff
@@ -35,6 +35,13 @@ func (br PodDecorations) Swap(i, j int) {
 }
 
 func lessPD(a, b *appsv1alpha1.PodDecoration) bool {
+	if a == nil || a.Spec.Weight == nil {
+		return true
```

Comment:
Since `weight` is 0 by default, if weight is not set, it should return false.

Reply:
Yes. Fixed.